src/Engine/Component/GeometryComponent.cpp

```diff
@@ -217,59 +223,70 @@ void TriangleMeshComponent::setupIO( const std::string& id ) {
         getEntity(), this, id, roOut );
 
-    ComponentMessenger::CallbackTypes<Ra::Core::Vector3Array>::ReadWrite vRW =
-        std::bind( &TriangleMeshComponent::getVerticesRw, this );
-    ComponentMessenger::getInstance()->registerReadWrite<Ra::Core::Vector3Array>(
-        getEntity(), this, id + "v", vRW );
-
-    ComponentMessenger::CallbackTypes<Ra::Core::Vector3Array>::ReadWrite nRW =
-        std::bind( &TriangleMeshComponent::getNormalsRw, this );
-    ComponentMessenger::getInstance()->registerReadWrite<Ra::Core::Vector3Array>(
-        getEntity(), this, id + "n", nRW );
-
-    ComponentMessenger::CallbackTypes<TriangleArray>::ReadWrite tRW =
-        std::bind( &TriangleMeshComponent::getTrianglesRw, this );
-    ComponentMessenger::getInstance()->registerReadWrite<TriangleArray>(
-        getEntity(), this, id + "t", tRW );
+    // ComponentMessenger::CallbackTypes<
+    //     Ra::Core::Geometry::TriangleMesh::PointAttribHandle::AttribType>::ReadWrite vRW =
+    //     std::bind( &TriangleMeshComponent::getVerticesRw, this );
+    // ComponentMessenger::getInstance()
+    //     ->registerReadWrite<Ra::Core::Geometry::TriangleMesh::PointAttribHandle::AttribType>(
+    //         getEntity(), this, id + "v", vRW );
+
+    // ComponentMessenger::CallbackTypes<
+    //     Ra::Core::Geometry::TriangleMesh::PointAttribHandle::AttribType>::ReadWrite nRW =
+    //     std::bind( &TriangleMeshComponent::getNormalsRw, this );
+    // ComponentMessenger::getInstance()
+    //     ->registerReadWrite<Ra::Core::Geometry::TriangleMesh::PointAttribHandle::AttribType>(
+    //         getEntity(), this, id + "n", nRW );
+
+    //    ComponentMessenger::CallbackTypes<TriangleArray>::ReadWrite tRW =
+    //        std::bind( &TriangleMeshComponent::getTrianglesRw, this );
+    //    ComponentMessenger::getInstance()->registerReadWrite<TriangleArray>(
+    //        getEntity(), this, id + "t", tRW );
 }
 
 const Ra::Core::Geometry::TriangleMesh* TriangleMeshComponent::getMeshOutput() const {
-    return &m_displayMesh->getTriangleMesh();
+    return &m_displayMesh->getCoreGeometry();
 }
 
 Ra::Core::Geometry::TriangleMesh* TriangleMeshComponent::getMeshRw() {
     CORE_ASSERT( m_displayMesh != nullptr, "DisplayMesh should exist while component is alive" );
-    m_displayMesh->setDirty( Mesh::VERTEX_POSITION );
-    m_displayMesh->setDirty( Mesh::VERTEX_NORMAL );
-    m_displayMesh->setDirty( Mesh::INDEX );
-    m_displayMesh->setDirty( Mesh::VERTEX_TANGENT, true );
-    m_displayMesh->setDirty( Mesh::VERTEX_BITANGENT, true );
-    m_displayMesh->setDirty( Mesh::VERTEX_TEXCOORD, true );
-    m_displayMesh->setDirty( Mesh::VERTEX_COLOR, true );
-    m_displayMesh->setDirty( Mesh::VERTEX_WEIGHTS, true );
-    m_displayMesh->setDirty( Mesh::VERTEX_WEIGHT_IDX, true );
-    return &( m_displayMesh->getTriangleMesh() );
+    ///\todo check if it's ok ;) but I think we do note need this anymore
```

Comment:
what about this todo?

Reply:
well it's a todo, it works for me, need feedback from others.

Reply:
humm maybe this one is outdated

Reply:
then remove the method and related commented dead code